src/iceberg/test/update_sort_order_test.cc

```diff
@@ -0,0 +1,272 @@
+
+  update->AddSortField(std::move(term), SortDirection::kAscending, NullOrder::kFirst);
+  EXPECT_THAT(update->Commit(), IsOk());
+
+  // Verify the sort order was set
+  ICEBERG_UNWRAP_OR_FAIL(auto reloaded, catalog_->LoadTable(table_ident_));
+  ICEBERG_UNWRAP_OR_FAIL(auto sort_order, reloaded->metadata()->SortOrder());
+  ASSERT_NE(sort_order, nullptr);
+  EXPECT_EQ(sort_order->fields().size(), 1);
+
+  const auto& field = sort_order->fields()[0];
+  EXPECT_EQ(field.source_id(), 1);
+  EXPECT_EQ(field.direction(), SortDirection::kAscending);
+  EXPECT_EQ(field.null_order(), NullOrder::kFirst);
+}
+
+TEST_F(UpdateSortOrderTest, AddSingleSortFieldDescending) {
+  ICEBERG_UNWRAP_OR_FAIL(auto update, table_->NewUpdateSortOrder());
+  auto ref = NamedReference::Make("y").value();
+  auto term = UnboundTransform::Make(std::move(ref), Transform::Identity()).value();
+
+  update->AddSortField(std::move(term), SortDirection::kDescending, NullOrder::kLast);
+  EXPECT_THAT(update->Commit(), IsOk());
+
+  // Verify the sort order was set
+  ICEBERG_UNWRAP_OR_FAIL(auto reloaded, catalog_->LoadTable(table_ident_));
+  ICEBERG_UNWRAP_OR_FAIL(auto sort_order, reloaded->metadata()->SortOrder());
+  ASSERT_NE(sort_order, nullptr);
+  EXPECT_EQ(sort_order->fields().size(), 1);
+
+  const auto& field = sort_order->fields()[0];
+  EXPECT_EQ(field.source_id(), 2);
+  EXPECT_EQ(field.direction(), SortDirection::kDescending);
+  EXPECT_EQ(field.null_order(), NullOrder::kLast);
+}
+
+TEST_F(UpdateSortOrderTest, AddMultipleSortFields) {
+  ICEBERG_UNWRAP_OR_FAIL(auto update, table_->NewUpdateSortOrder());
+  auto ref1 = NamedReference::Make("y").value();
+  auto term1 = UnboundTransform::Make(std::move(ref1), Transform::Identity()).value();
+
+  auto ref2 = NamedReference::Make("x").value();
+  auto term2 = UnboundTransform::Make(std::move(ref2), Transform::Identity()).value();
+
+  update->AddSortField(std::move(term1), SortDirection::kAscending, NullOrder::kFirst)
+      .AddSortField(std::move(term2), SortDirection::kDescending, NullOrder::kLast);
+
+  EXPECT_THAT(update->Commit(), IsOk());
+
+  // Verify the sort order was set with multiple fields
+  ICEBERG_UNWRAP_OR_FAIL(auto reloaded, catalog_->LoadTable(table_ident_));
+  ICEBERG_UNWRAP_OR_FAIL(auto sort_order, reloaded->metadata()->SortOrder());
+  ASSERT_NE(sort_order, nullptr);
+  EXPECT_EQ(sort_order->fields().size(), 2);
+
+  // Check first field (y field)
+  const auto& field1 = sort_order->fields()[0];
+  EXPECT_EQ(field1.source_id(), 2);
+  EXPECT_EQ(field1.direction(), SortDirection::kAscending);
+  EXPECT_EQ(field1.null_order(), NullOrder::kFirst);
+
+  // Check second field (x field)
+  const auto& field2 = sort_order->fields()[1];
+  EXPECT_EQ(field2.source_id(), 1);
+  EXPECT_EQ(field2.direction(), SortDirection::kDescending);
+  EXPECT_EQ(field2.null_order(), NullOrder::kLast);
+}
+
+TEST_F(UpdateSortOrderTest, AddSortFieldWithTruncateTransform) {
```

Comment:
These cases look so similar. Perhaps use a common function for different inputs?